Source/DafnyDriver/DafnyDriver.cs

```diff
@@ -505,6 +505,36 @@ public static CommandLineArgumentsResult ProcessCommandLineArguments(string[] ar
 
     class DafnyConsolePrinter : ConsolePrinter
     {
+      private readonly Dictionary<string, List<string>> fsCache = new Dictionary<string, List<string>>();
+
+      private string GetFileLine(string filename, int lineNumber) {
+        List<string> lines;
+        if (!fsCache.ContainsKey(filename)) {
+          // Note: This is not guaranteed to be the same file that Dafny parsed. To ensure that, Dafny should keep
+          // an in-memory version of each file it parses.
+          lines = File.ReadLines(filename).ToList();
```

Comment:
This seems expensive for large files with many errors.

Reply:
The size of the file matters, yes, but the number of errors should not.

Reply:
O, sorry I skimmed over the cache.